tests/test_array_spec.py

```diff
@@ -0,0 +1,282 @@
+import unittest
+
+from json_converter.json_mapper import JsonMapper
+from json_converter.post_process import default_to
+
+
+def array_to_string(*args):
+    value = ", ".join(args[0])
+    return value
+
+
+class TestMappingArrays(unittest.TestCase):
+    @unittest.skip("not supported yet")
+    def test_source_array_on(self):
+        expected_json = {
+            "authors": ["Stephenson E", "Niel Tyson"]
+        }
+        json_object = {
+            "publications": [
+                {
+                  "author": "Stephenson E"
+                },
+                {
+                    "author": "Niel Tyson"
+                }
+            ]
+        }
+        test_spec = {
+            "$on": "publications",
+            "authors": ["$array", ["author"], True]
+        }
+        authors_json = JsonMapper(json_object).map(test_spec)
+        self.assertDictEqual(expected_json, authors_json)
+
+    @unittest.skip("not supported yet")
```

Comment:
Will support for enabling these tests be added in this PR or in a future ticket?

Reply:
TBH I was thinking about what to do with these tests.
We don't need this functionality for now.
@MightyAx Shall I remove them from this PR or as you suggested create a ticket to support them later when we need it?

Reply:
It's okay to just retain these tests, the ticket can be created when we actually need it.

Reply:
Because of the way the current spec documents are structured there isn't a way to fulfil the skipped tests. The `$array` command maps each dictionary in the array to the same spec, an array of 5 dictionaries is still an array of 5 dictionaries just with mapped content.

```example.py
input = {
    "fiction": [
        {
            "book_title": "Axioms End",
            "book_author": "Lindsay Ellis"
        },
        {
            "book_title": "The Starless Sea",
            "book_author": "Erin Morgenstern"
        }
    ],
    "fact": [
        {
            "book_title": "Yes Man",
            "book_author": "Danny Wallace"
        }
    ]
}
spec = {
    "books": ["$array", {'$on': 'fiction', 'title': ['book_title']}, True]
}
expected = {
    "books": [
        {'title': "Axioms End"},
        {'title': "The Starless Sea"}
    ]
}
```

The skipped unit tests are trying to convert the array of 5 dictionaries to an array of 5 strings, so we will need the user to specify more information that the current command captures in order to get:
```
expected = {
    "books": ["Axioms End", "The Starless Sea"]
}
```
I think this requirement is possible to satisfy, but not without a some new command language that the unit tests can not anticipate which make the usefulness very low

Reply:
I removed these tests as discussed previously